x/evm/types/state_object.go

```diff
@@ -400,7 +400,16 @@ func (so *stateObject) GetCommittedState(_ ethstate.Database, key ethcmn.Hash) e
 func (so *stateObject) ReturnGas(gas *big.Int) {}
 
 func (so *stateObject) deepCopy(db *CommitStateDB) *stateObject {
-	newStateObj := newStateObject(db, so.account)
+	newAccount := ethermint.ProtoAccount().(*ethermint.EthAccount)
+	jsonAccount, err := so.account.MarshalJSON()
+	if err != nil {
+		return nil
```

Comment:
seems that using value copy is ok here but when error happens during `marshal` or `unmarshal`, error will be suppressed and nil will be returned, in this case, any one who invoked `deepCopy` may encounter `nil pointer dereference` and there is no way to prevent it.

Reply:
The error returned in marshal is usually due to an operating system error. In this case, panic may be more appropriate.

Reply:
I wrote another solution, you can take a look, but I am not familiar with the usage of reflect, and I am worried that there will be any problems.

Reply:
@freddyli7 Looking forward to your reply